### DISCUSSION TITLE
Create an entity after destroying one

### DISCUSSION THREAD

**Discussion by gamecoder-nz** [skypjack/entt#1120] [Q&A] [answered]
I have this code
` entt::entity entity = registry.create();
 registry.destroy(entity);
 entt::entity entity2 = registry.create();`

but entity2 is 1048576.  Is this correct?

**Comment by skypjack** [comment, marked as answer]
> Is this correct?

Well, yeah, definitely. It would be a huge problem otherwise. 🙂 
When you destroy an identifier, it's recycled with a different version number. Recycled identifiers are returned before creating new ones.
If you look the entity part of the identifier, it's the same as before. However, the version changes from 0 to 1. That's all.
